lib/app_server.py

```diff
@@ -28,6 +28,10 @@ def run_server(execs, cwd, server, logfile, retval):
 class AppTest(Test):
     def execute(self, server):
         server.current_test = self
+        if self.suite_ini.get('pre_cleanup', False):
```

Comment:
It would be good to do the check just like `if self.suite_ini['pre_cleanup']:` if possible.